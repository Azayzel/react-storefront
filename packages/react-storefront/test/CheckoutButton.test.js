/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import CheckoutButton from '../src/CheckoutButton'
import TestProvider from './TestProvider'
import { createMemoryHistory } from 'history'

describe('CheckoutButton', () => {
  it('renders', () => {
    const component = (
      <TestProvider>
        <CheckoutButton url="/checkout" />
      </TestProvider>
    )
    expect(mount(component)).toMatchSnapshot()
  })

  it('navigates via history when configured with a path', () => {
    const history = createMemoryHistory()
    history.push = jest.fn()

    const component = (
      <TestProvider history={history}>
        <CheckoutButton url="/checkout" />
      </TestProvider>
    )

    mount(component)
      .find('button')
      .simulate('click')

    expect(history.push).toHaveBeenCalledWith('/checkout', undefined)
  })

  it('navigates via window location change when configured with a full url', () => {
    const setter = jest.fn()
    const { location } = window

    try {
      delete window.location
      window.location = {}

      Object.defineProperty(window.location, 'href', {
        get: () => 'http://localhost',
        set: setter
      })

      const component = (
        <TestProvider history={history}>
          <CheckoutButton url="http://localhost/checkout" />
        </TestProvider>
      )

      mount(component)
        .find('button')
        .simulate('click')

      expect(setter).toHaveBeenCalledWith('http://localhost/checkout')
    } finally {
      window.location = location
    }
  })
})
