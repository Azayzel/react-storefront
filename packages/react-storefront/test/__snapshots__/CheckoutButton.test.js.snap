// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutButton renders 1`] = `
<TestProvider>
  <Provider router={{...}} app={{...}} history={{...}} nextId={[Function: nextId]}>
    <MuiThemeProviderOld theme={{...}}>
      <WithStyles(inject-CheckoutButton-with-router) url="/checkout">
        <inject-CheckoutButton-with-router url="/checkout" classes={{...}}>
          <CheckoutButton url="/checkout" classes={{...}} router={{...}} docked={false}>
            <WithStyles(Button) size="large" color="secondary" onClick={[Function]} variant="contained" data-th="checkout-link" router={{...}} classes={{...}} className="">
              <Button size="large" color="secondary" onClick={[Function]} variant="contained" data-th="checkout-link" router={{...}} className="" classes={{...}} component="button" disabled={false} disableFocusRipple={false} fullWidth={false} mini={false} type="button">
                <WithStyles(ButtonBase) className="MuiButton-root-3 RSFCheckoutButton-root-1 MuiButton-contained-14 MuiButton-containedSecondary-16 MuiButton-raised-17 MuiButton-raisedSecondary-19 MuiButton-sizeLarge-27" disabled={false} focusRipple={true} focusVisibleClassName="MuiButton-focusVisible-22" onClick={[Function]} data-th="checkout-link" router={{...}} component="button" type="button">
                  <ButtonBase className="MuiButton-root-3 RSFCheckoutButton-root-1 MuiButton-contained-14 MuiButton-containedSecondary-16 MuiButton-raised-17 MuiButton-raisedSecondary-19 MuiButton-sizeLarge-27" disabled={false} focusRipple={true} focusVisibleClassName="MuiButton-focusVisible-22" onClick={[Function]} data-th="checkout-link" router={{...}} component="button" type="button" classes={{...}} centerRipple={false} disableRipple={false} disableTouchRipple={false} tabIndex="0">
                    <button className="MuiButtonBase-root-29 MuiButton-root-3 RSFCheckoutButton-root-1 MuiButton-contained-14 MuiButton-containedSecondary-16 MuiButton-raised-17 MuiButton-raisedSecondary-19 MuiButton-sizeLarge-27" onBlur={[Function]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex="0" type="button" disabled={false} onClick={[Function]} data-th="checkout-link" router={{...}}>
                      <span className="MuiButton-label-4">
                        Checkout
                      </span>
                      <NoSsr defer={false} fallback={{...}}>
                        <WithStyles(TouchRipple) innerRef={[Function]} center={false}>
                          <TouchRipple center={false} classes={{...}}>
                            <TransitionGroup component="span" enter={true} exit={true} className="MuiTouchRipple-root-32" childFactory={[Function: childFactory]}>
                              <span className="MuiTouchRipple-root-32" />
                            </TransitionGroup>
                          </TouchRipple>
                        </WithStyles(TouchRipple)>
                      </NoSsr>
                    </button>
                  </ButtonBase>
                </WithStyles(ButtonBase)>
              </Button>
            </WithStyles(Button)>
          </CheckoutButton>
        </inject-CheckoutButton-with-router>
      </WithStyles(inject-CheckoutButton-with-router)>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;
